teps/0060-remote-resource-resolution.md

```diff
@@ -0,0 +1,386 @@
+   with an org's chosen storage. Put another way: [why can't we just keep our
+   Tasks in git?](https://github.com/tektoncd/pipeline/issues/2298)
+3. Pipeline authors have to document the Tasks their Pipeline depends on
+   out-of-band. A [common example of
+   this](https://github.com/tektoncd/pipeline/discussions/3822) is a build
+   Pipeline that needs `git-clone` to be installed from the open source catalog
+   before the Pipeline can be run.
+   [TEP-0053](https://github.com/tektoncd/community/pull/352) is also exploring
+   ways of encoding this information as part of pipelines in the catalog.
+4. Tutorials introducing Tekton Pipelines have to include instructions for
+   fetching and manually installing tasks that the tutorial relies on.
+5. Pipelines' [resolver
+   code](https://github.com/tektoncd/pipeline/blob/bad45506a9a9c25b028703e85e501b8f97598695/pkg/reconciler/taskrun/resources/taskref.go#L38)
+   is synchronous: a reconciler thread is blocked while a resource is being
+   fetched.
+6. Pipelines' existing support for cluster and bundle resources is not easily
+   extensible without modifying Pipelines' source code.
+
+### Goals
+
+- Aspirationally, never require a user to run `kubectl apply -f git-clone.yaml`
+  ever again.
+- Provide the ability to manage Tekton resources in an org's source of choice
+  via any process of their choice - `git push`, `tkn bundle push`, `gsutil cp`
+  and so on, instead of `kubectl apply`.
+- Allow multiple remotes to be supported in a single pipeline: task A from OCI,
+  task B from the Catalog, task C directly from the cluster, and so on.
+- Allow the open source Pipelines project (and any downstream project) to
+  choose which remotes are enabled out of the box with a new release.
+- Allow operators to add new remote sources without having to redeploy the
+  entirety of Tekton Pipelines.
+- Allow operators to completely remove the code paths that fetch Tekton
+  resources from remotes they don't want to support.
+- Allow operators to control the threshold at which fetching a remote Tekton
+  resource is considered timed out.
+- Emit Events and Conditions operators can use to assess whether fetching
+  remote resources is slow or failing.
+- Establish a common syntax that tool and platform creators can use to record,
+  as part of a pipeline or run, the remote location that Tekton resources
+  should be fetched from.
+- Offer a mechanism to verify remote tasks and pipelines against a digest
+  (or similar mechanism) before they are processed as resources by Pipelines'
+  reconcilers to ensure that the resource returned by a given Remote matches the
+  resource expected by the user.
+
+### Non-Goals
+
+### Use Cases (optional)
+
+**Sharing a git repo of tasks**: An organization decides to manage their
+internal tasks and pipelines in a shared git repository. The CI/CD cluster is
+set up to allow these resources to be referenced directly by TaskRuns and
+PipelineRuns in the cluster.
```

Comment:
If we dig deeper on this one, how would this work ? The `PipelineRun` has a `commit` sha as parameter and use that parameter to refer to a given `Pipeline` right ? But then, to be *sure* to refer to the correct `Task` from this `Pipeline`, we need the same `commit` sha parameter, isn't it ? (so that all task are refered using this particular `commit` sha)

Reply:
Again, also seems like an implementation detail but in general, maybe? It doesn't have to be a parameter, could be part of the resolver url eg `github.com/tektoncd/community@sha:foo-bar`. I just made this off the top of my head but the salient bit is that it is on the developer of the resolver to specify the format of the input and the contract they have is with the end user to provide that format.

Tekton's role in all this is to provide the API machinery for sending that input to a resolver and expecting the resource back.